node/db/src/parity_db.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright 2019-2022 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use super::errors::Error;
+use crate::utils::bitswap_missing_blocks;
+use crate::Store;
+use cid::Cid;
+use fvm_ipld_blockstore::Blockstore;
+use libp2p_bitswap::BitswapStore;
+use parity_db::Db;
+use parity_db::Options;
+use std::collections::HashMap;
+use std::path::Path;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+#[derive(Clone)]
+pub struct ParityDb {
+    pub db: Arc<parity_db::Db>,
+}
+
+pub struct ParityDbConfig {
+    pub path: PathBuf,
+    pub columns: u8,
+}
+
+impl ParityDbConfig {
+    pub fn from_path(path: &Path) -> Self {
+        Self {
+            path: path.to_path_buf(),
+            columns: 1,
+        }
+    }
+}
+
+impl ParityDb {
+    fn to_options(config: &ParityDbConfig) -> Options {
+        Options {
+            path: config.path.to_owned(),
+            sync_wal: true,
+            sync_data: true,
+            stats: true,
+            salt: None,
+            columns: (0..config.columns)
+                .map(|_| parity_db::ColumnOptions {
+                    compression: parity_db::CompressionType::Lz4,
+                    ..Default::default()
+                })
+                .collect(),
+            compression_threshold: HashMap::new(),
+        }
+    }
+
+    pub fn open(config: &ParityDbConfig) -> anyhow::Result<Self> {
+        let opts = Self::to_options(config);
+        Ok(Self {
+            db: Arc::new(Db::open_or_create(&opts)?),
+        })
+    }
+}
+
+impl Store for ParityDb {
+    fn read<K>(&self, key: K) -> Result<Option<Vec<u8>>, Error>
+    where
+        K: AsRef<[u8]>,
+    {
+        self.db.get(0, key.as_ref()).map_err(Error::from)
+    }
+
+    fn write<K, V>(&self, key: K, value: V) -> Result<(), Error>
+    where
+        K: AsRef<[u8]>,
+        V: AsRef<[u8]>,
+    {
+        let tx = [(0, key.as_ref(), Some(value.as_ref().to_owned()))];
```

Comment:
Maybe implement `bulk_write` as well? Car import may benefit from it.

Reply:
> Very nice!
> 
>     * What settings did you took for RocksDB? (The last one for main, so the aggressive optim? just want to make sure here what we are comparing).
I believe it was the last default config from `main` from yesterday, as I was pretty surprised by how fast it was when compared to my previous runs. :)

>     * What is the db size with compression for ParityDb? Is it as good as for RocksDB?
>

So far I'm not impressed with the compression, it sits on 92GB, so it's worse than RocksDB with the current settings.
 
>     * What does the "run" command do?
It's just running Forest for 1h after the snapshot is already imported, i.e. `target/release/forest --target-peer-count 50 --encrypt-keystore false`

Reply:
Splendid idea!